app/controllers/remove/PsaRemovePspDeclarationController.scala

```diff
@@ -78,22 +98,37 @@ class PsaRemovePspDeclarationController @Inject()(
               form.bindFromRequest().fold(
                 (formWithErrors: Form[Boolean]) =>
                   Future.successful(BadRequest(view(formWithErrors, schemeName, srn, index))),
-                value =>
-                  userAnswersCacheConnector.save(request.externalId, PsaRemovePspDeclarationId(index), value).flatMap {
-                    cacheMap =>
-                      pspConnector.deAuthorise(
-                        pstr = pstr,
-                        deAuthorise = DeAuthorise(
-                          ceaseIDType = "PSP",
-                          ceaseNumber = pspDetails.id,
-                          initiatedIDType = "PSA",
-                          initiatedIDNumber = request.psaIdOrException.id,
-                          ceaseDate = LocalDate.now().toString
+                value => {
+                    for {
+                      cacheMap <- userAnswersCacheConnector.save(request.externalId, PsaRemovePspDeclarationId(index), value)
+                      _ <-  pspConnector.deAuthorise(
+                          pstr = pstr,
+                          deAuthorise = DeAuthorise(
+                            ceaseIDType = "PSP",
+                            ceaseNumber = pspDetails.id,
+                            initiatedIDType = "PSA",
+                            initiatedIDNumber = request.psaIdOrException.id,
+                            ceaseDate = LocalDate.now().toString
                         )
-                      ).map {
-                        _ =>
-                          Redirect(navigator.nextPage(PsaRemovePspDeclarationId(index), NormalMode, UserAnswers(cacheMap)))
+                      )
+                      minimalDetails <- minimalPsaConnector.getMinimalPsaDetails(request.psaIdOrException.id)
+                      emailResponse <- emailConnector.sendEmail(
+                          SendEmailRequest(
+                            to = List(minimalDetails.email),
+                            templateId = appConfig.emailPsaDeauthorisePspTemplateId,
+                            parameters = Map(
+                              "psaName" -> minimalDetails.name,
+                              "pspName" -> pspDetails.name,
+                              "schemeName" -> schemeName
+                            )
+                          )
+                        )
+                    } yield {
+                      if (emailResponse== EmailNotSent) {
+                        Logger.error("Unable to send email to de-authorising PSA. Support intervention possibly required.")
                       }
+                      Redirect(navigator.nextPage(PsaRemovePspDeclarationId(index), NormalMode, UserAnswers(cacheMap)))
+                    }
                   }
               )
             } else {
```

Comment:
else that has nothing in it?

Reply:
It goes to the session expired controller in case of the else